### PR TITLE
Refactor RB_ADD_PYTESTBENCHMARK and add RDF tutorial df102 as benchmark

### PR DESCRIPTION
**Pull request by stwunsch**
Tadaa! :tada: 

@eguiraud @oshadura Please review! I may have done stupid things, although everything works fine for me :)

### CODE REVIEW

File: cmake/modules/AddRootBench.cmake [hunk 1]
```diff
@@ -60,25 +78,42 @@ endfunction(RB_ADD_GBENCHMARK)
...
+      if(ARG_SETUP)
+         RB_ADD_SETUP_FIXTURE(${benchmark} SETUP ${ARG_SETUP})
+      endif()
+
+      file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/${filename} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/${benchmark} USE_SOURCE_PERMISSIONS)
+
+      add_test(NAME rootbench-${benchmark}
+               COMMAND ${PYTHON_EXECUTABLE} -B -m pytest ${CMAKE_CURRENT_SOURCE_DIR}/${filename} -v --csv rootbench-pytest-${benchmark}.csv)
+      set_tests_properties(rootbench-${benchmark} PROPERTIES
+                           ENVIRONMENT "LD_LIBRARY_PATH=${ROOT_LIBRARY_DIR}:$ENV{LD_LIBRARY_PATH};PYTHONPATH=${ROOTSYS}/lib:$ENV{PYTHONPATH};RB_DATASETDIR=${RB_DATASETDIR}"
```
Comment (eguiraud) [inline cmake/modules/AddRootBench.cmake:110]: the `RB::GetDataDir` helper function is not available from python?

File: cmake/modules/AddRootBench.cmake [hunk 2]
```diff
@@ -60,25 +78,42 @@ endfunction(RB_ADD_GBENCHMARK)
...
+                           TIMEOUT "${TIMEOUT_VALUE}" LABELS "${ARG_LABEL}" RUN_SERIAL TRUE
+                           FIXTURES_REQUIRED "setup-${benchmark};download-${benchmark}-datafiles")
+    else()
+      message(STATUS "ROOT was configured without PyROOT support. Python benchmarks will be disabled!")
```
Comment (eguiraud) [inline cmake/modules/AddRootBench.cmake:114]: if I read correctly indentation of this message is now wrong and it's executed if `NOT PYTEST_FOUND`

File: cmake/modules/AddRootBench.cmake (hunk 1 above)
Comment (stwunsch) [reply cmake/modules/AddRootBench.cmake:110]: It's a header only thingy. I could probably load somehow the header, but an envvar is just simpler and nicer.

**Comment by oshadura** [comment]
@eguiraud are we are good here?

File: cmake/modules/AddRootBench.cmake (hunk 2 above)
Comment (stwunsch) [reply cmake/modules/AddRootBench.cmake]: You are right, fixed!

**Comment by stwunsch** [comment]
Fixed the logic if pytest is not found! The current behaviour is that we don't fail but just do not add the test. What do we want?

**Comment by stwunsch** [comment]
Rebased! @eguiraud @oshadura Good to go?

File: cmake/modules/AddRootBench.cmake [hunk 3]
```diff
@@ -60,23 +78,42 @@ endfunction(RB_ADD_GBENCHMARK)
...
+      file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/${filename} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/${benchmark} USE_SOURCE_PERMISSIONS)
+
+      add_test(NAME rootbench-${benchmark}
+               COMMAND ${PYTHON_EXECUTABLE} -B -m pytest ${CMAKE_CURRENT_SOURCE_DIR}/${filename} -v --csv rootbench-pytest-${benchmark}.csv)
+      set_tests_properties(rootbench-${benchmark} PROPERTIES
+                           ENVIRONMENT "LD_LIBRARY_PATH=${ROOT_LIBRARY_DIR}:$ENV{LD_LIBRARY_PATH};PYTHONPATH=${ROOTSYS}/lib:$ENV{PYTHONPATH};RB_DATASETDIR=${RB_DATASETDIR}"
+                           TIMEOUT "${TIMEOUT_VALUE}" LABELS "${ARG_LABEL}" RUN_SERIAL TRUE
+                           FIXTURES_REQUIRED "setup-${benchmark};download-${benchmark}-datafiles")
+    else()
+      message(STATUS "pytest was not found, don't add Python benchmark" ${benchmark})
```
Comment (eguiraud) [inline cmake/modules/AddRootBench.cmake:114]: very minor thing, the imperative mood is weird here

Comment (stwunsch) [reply cmake/modules/AddRootBench.cmake:114]: Feel free to make a proposal through the "suggest change" feature of github! :)

Comment (stwunsch) [inline cmake/modules/AddRootBench.cmake:114]:
```suggestion
      message(STATUS "pytest was not found, benchmark" ${benchmark} " will be ignored")
```

**Pull request merged**